src/sys/sys_unix.c

```diff
@@ -269,6 +269,15 @@ int Sys_Milliseconds(void)
 	return curtime;
 }
 
+/**
+ * @brief Sys_Microseconds
+ * @return
+ */
+int64_t Sys_Microseconds(void)
+{
+	return Sys_Milliseconds() * 1000;
```

Comment:
This is more of a note (I can implement it after the PR is merged). The Unix side also needs to have the Microseconds function to the actual time calc and the milliseconds just needs to return that value subdivided by a 1000.